graviti/utility/requests.py

```diff
@@ -0,0 +1,254 @@
+            **{_ALLOWED_METHODS: config.allowed_retry_methods},  # type: ignore[arg-type]
+        )
+
+        self.mount("http://", TimeoutHTTPAdapter(20, 20, retry_strategy))
+        self.mount("https://", TimeoutHTTPAdapter(20, 20, retry_strategy))
+
+    def request(  # type: ignore[override]
+        self, method: str, url: str, *args: Any, **kwargs: Any
+    ) -> Response:  # noqa: DAR401
+        """Make the request.
+
+        Arguments:
+            method: Method for the request.
+            url: URL for the request.
+            *args: Extra arguments to make the request.
+            **kwargs: Extra keyword arguments to make the request.
+
+        Returns:
+            Response of the request.
+
+        Raises:
+            ResponseError: If post response error.
+
+        """
+        try:
+            response = super().request(method, url, *args, **kwargs)
+            if response.status_code not in (200, 201):
+                logger.error(
+                    "Unexpected status code(%d)!%s", response.status_code, ResponseLogging(response)
+                )
+                raise ResponseError(response=response)
+
+            logger.debug(ResponseLogging(response))
+            return response
+
+        except RequestException as error:
+            logger.error(
+                "%s.%s: %s%s",
+                error.__class__.__module__,
+                error.__class__.__name__,
+                error,
+                RequestLogging(error.request),
+            )
+            raise
+
+
+SESSIONS: DefaultDict[int, UserSession] = defaultdict(UserSession)
+
+
+def get_session() -> UserSession:
+    """Create and return a session per PID so each sub-processes will use their own session.
+
+    Returns:
+        The session corresponding to the process.
+    """
+    return SESSIONS[os.getpid()]
+
+
+class UserResponse:
+    """This class used to read data from Response with stream method.
+
+    Arguments:
+        response: Response of the Session.request().
+
+    """
+
+    def __init__(self, response: Response):
+        self.response = response
+
+    def __enter__(self) -> "UserResponse":
+        return self
+
+    def __exit__(self, *_: Any) -> None:
+        self.close()
+
+    def close(self) -> None:
+        """Close the response."""
+        self.response.close()
+
+    def read(self, amt: Optional[int] = None) -> bytes:
+        """Read data from response.
+
+        Arguments:
+            amt: The needed read amount.
+
+        Returns:
+            Response of the request.
+
+        """
+        if amt is None:
+            try:
+                return b"".join(chunk for chunk in self.response.iter_content(_CHUNK_SIZE))
+
+            except StreamConsumedError:
+                return b""
+
+        try:
+            return next(self.response.iter_content(amt))  # type: ignore[no-any-return]
+
+        except StopIteration:
+            return b""
+
+
+class Tqdm(tqdm):  # type: ignore[misc]
```

Comment:
`Tqdm` is not used currently. Remove it for now?

Reply:
ok